python/lsst/afw/geom/coordinates/coordinates.cc

```diff
@@ -0,0 +1,509 @@
+    cls.def("setX", [](Extent<T, 3> & self, T other) { self[0] = other; });
+    cls.def("setY", [](Extent<T, 3> & self, T other) { self[1] = other; });
+    cls.def("setZ", [](Extent<T, 3> & self, T other) { self[2] = other; });
+
+    return cls;
+}
+
+// Declare mixed-type and type-overloaded operators for Extent with dimension
+// N for both int and double. Because pybind11 tries operators (like any
+// overload) `in order', int has to come before double in any overloaded
+// operators that dispatch on a scalar, and hence they have to be defined here
+// instead of declareExtent.
+template <int N>
+void declareExtentOperators(py::module & mod, PyExtent<int,N> & clsI, PyExtent<double,N> & clsD) {
+    // Python's integer division works differently than C++'s for negative numbers - Python
+    // uses floor (rounds towards more negative), while C++ truncates (rounds towards zero).
+    // Therefore one needs to be careful in the definition of division operators.
+    clsI.def("__floordiv__",
+             [](Extent<int, N> const &self, int other) -> Extent<int, N> {
+                 return floor(self / static_cast<double>(other));
+             },
+             py::is_operator());
+
+    clsI.def("__div__",
+             [](Extent<int, N> const &self, int other) { return floor(self / static_cast<double>(other)); },
+             py::is_operator());
+    clsI.def("__div__", [](Extent<int, N> const &self, double other) { return self / other; },
+             py::is_operator());
+    clsD.def("__div__", [](Extent<double, N> const &self, double other) { return self / other; },
+             py::is_operator());
+
+    clsI.def("__truediv__", [](Extent<int, N> const &self, double other) { return self / other; },
+             py::is_operator());
+    clsD.def("__truediv__", [](Extent<double, N> const &self, double other) { return self / other; },
+             py::is_operator());
+
+    clsI.def("__ifloordiv__", [](Extent<int, N> &self, int other) -> Extent<int, N> & {
+        self = floor(self / static_cast<double>(other));
+        return self;
+    });
+
+    clsI.def("__idiv__", [](Extent<int, N> &self, int other) -> Extent<int, N> & {
+        self = floor(self / static_cast<double>(other));
+        return self;
+    });
+    clsD.def("__idiv__", [](Extent<double, N> &self, double other) -> Extent<double, N> & {
+        self /= other;
+        return self;
+    });
+
+    clsI.def("__itruediv__", [](Extent<int, N> &self, double other) {
+        PyErr_SetString(PyExc_TypeError, "In-place true division not supported for Extent<int,N>.");
+        throw py::error_already_set();
+    });
+    clsD.def("__itruediv__", [](Extent<double, N> &self, double other) -> Extent<double, N> & {
+        self /= other;
+        return self;
+    });
+
+    clsI.def("__iadd__", [](Extent<int, N> &self, Extent<int, N> const &other) -> Extent<int, N> & {
+        self += other;
+        return self;
+    });
```

Comment:
This is giving me a feeling of deja vu, but what's our convention on using lambdas vs. casted method pointers?